### ISSUE TITLE
Adding geometries does not work from touchscreens

### ISSUE THREAD

**Issue by wonderchook** [Cadasta/cadasta-platform#299]
Adding geometries such as in the "Add Location" screen or the "Project Location" screen does not work. You can zoom on the map, but if the user selects one of the options to draw a geometry it is not possible to draw the geometry using the touch screen. 

@seav can you look into this and propose a solution? If it is going to be time consuming we'll put it into the next release.

**Comment by oliverroick** [comment]
`django-leaflet` uses `Leaflet.Draw` for editing geometries. At UCL we had similar problems with `Leaflet.Draw` and touch screens; we were not able to fix it. We eventually replaced `Leaflet.Draw` with `Leaflet.Editable`, but I'm not sure if plays with `django-leaflet`.

**Comment by seav** [comment]
This is a known issue with Leaflet.draw:
https://github.com/Leaflet/Leaflet.draw/issues/2

Somebody updated the code and provided a PR to make Leaflet.draw support touch event: https://github.com/Leaflet/Leaflet.draw/pull/285

**Comment by seav** [comment]
OK. I learned that Leaflet.draw is a built-in plugin for django-leaflet. I would've liked to use a different JS file mentioned in the comment above to support touch events but I don't know how to override the built-in provided JS file. Would placing a file in `core/static/js` where the plugin's expected location work?

**Comment by seav** [comment]
OK. I saw that [Leaflet.draw](https://github.com/Leaflet/Leaflet.draw) has been updated to [version 0.3.0](https://github.com/Leaflet/Leaflet.draw/tree/v0.3.0) which has touch support. django-leaflet currently has a [PR](https://github.com/makinacorpus/django-leaflet/pull/153) to update the embedded Leaflet.draw from v0.2.4 to v0.3.0. I suggest we wait for this PR to be included and then update our django-leaflet dependency.

Anyway, I have tested Leaflet.draw v0.3.0 and can confirm that drawing works on touchscreens. Unfortunately, it seems that any drawn stuff disappears as soon as the drawing is done, whether by mouse or by finger. I think the API has changed and we need to update our code.

**Comment by wonderchook** [comment]
Pushed off until after next weeks release. I think this is a tolerable problem for the time being.

**Comment by wonderchook** [comment]
@seav I've pushed this off since the PR hasn't been merged yet, so we don't end up waiting on it.

**Comment by dpalomino** [comment]
How feasible/difficult would be to address this issue now? We have seen that is quite frequent in the field to use tablets to access to the platform, and also to draw location geometries.

**Comment by wonderchook** [comment]
looks like the pull request was closed without being merged. @seav do you think it would be feasible for us to update and make a pull request? Looks like they are willing to merge if not using a dev version. https://github.com/makinacorpus/django-leaflet/pull/153#issuecomment-253473200

**Comment by seav** [comment]
@wonderchook, it should be possible. But the dev version is a non-issue. That's how the maintainers of Leaflet.draw indicate the version in the JavaScript source code even if they have already tagged the code for release.

I'll submit a PR to update to the latest release which is 0.4.0.

**Comment by seav** [comment]
Ah, there's no need to submit a PR. There's already a [new PR](https://github.com/makinacorpus/django-leaflet/pull/169) to update to Leaflet 1.0 and Leaflet 0.4.0.

**Comment by dpalomino** [comment]
It is related with #1031, and upgrading django leaflet. To be re-tested.

**Comment by dpalomino** [comment]
Dup of #1127.
